exportReport/Export/ReportExport.java

```diff
@@ -0,0 +1,325 @@
+
+import org.apache.log4j.Logger;
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.core.scanner.Alert;
+import org.parosproxy.paros.extension.ViewDelegate;
+import org.parosproxy.paros.model.Model;
+import org.parosproxy.paros.model.SiteMap;
+import org.parosproxy.paros.model.SiteNode;
+import org.zaproxy.zap.extension.exportReport.json.JSONException;
+import org.zaproxy.zap.extension.exportReport.json.JSONObject;
+import org.zaproxy.zap.extension.exportReport.json.XML;
+import org.zaproxy.zap.extension.exportReport.model.AlertItem;
+import org.zaproxy.zap.extension.exportReport.model.Alerts;
+import org.zaproxy.zap.extension.exportReport.model.Report;
+import org.zaproxy.zap.extension.exportReport.model.Sites;
+import org.zaproxy.zap.utils.XMLStringUtil;
+import org.zaproxy.zap.view.ScanPanel;
+
+import org.w3c.dom.Document;
+import org.w3c.dom.ls.DOMImplementationLS;
+import org.w3c.dom.ls.LSSerializer;
+import org.xml.sax.SAXException;
+
+/*
+ * AUTHOR		:	GORAN SARENKAPA - JordanGS
+ * SPONSOR		:	RYERSON UNIVERSITY
+ * CLASS		:	ReportExport.java 
+ * DESC			:	Retrieves all the data, parses it into a report type and returns it back to ExportReport.java.
+ * CREATED ON	:	MARCH 10TH, 2016
+ * CURRENT VER	:	V1.0
+ * SOURCE		:	https://github.com/JordanGS/workspace/tree/master/zap-extensions/src/org/zaproxy/zap/extension/exportReport
+ */
+
+/* 
+ * MODIFED BY	:	<NAME> - <GIT USER>
+ * MOD DATE		:	
+ * MOD VERSION	:	<VERSION OF PLUGIN>
+ * MOD DESC		:	
+ */
+
+public class ReportExport {
+	
+	private static String entityEncode(String text) throws UnsupportedEncodingException {
+		String result = text;
+
+		if (result == null) {
+			return result;
+		}
+		// There is an encoding issue with the passed in String, this is a fix to maintain encoding and escapes!
+		byte ptext[] = result.getBytes("ISO-8859-1"); 
+		String value = new String(ptext, "UTF-8"); 
+		String temp = XMLStringUtil.escapeControlChrs(value);
+
+		return temp;
+	}
+
+	public static String generateDUMP(String path, String fileName, String reportTitle, String reportBy, String reportFor,
+			String scanDate, String scanVersion, String reportDate, String reportVersion, String reportDesc,
+			ArrayList<String> alertSeverity, ArrayList<String> alertDetails) throws UnsupportedEncodingException {
+
+		SiteMap siteMap = Model.getSingleton().getSession().getSiteTree();
+		SiteNode root = (SiteNode) siteMap.getRoot();
+		int siteNumber = root.getChildCount();
+
+		Report report = new Report();
+		report.setTitle(entityEncode(reportTitle));
+		report.setBy(entityEncode(reportBy));
+		report.setFor(entityEncode(reportFor));
+		report.setScanDate(entityEncode(scanDate));
+		report.setScanVersion(entityEncode(scanVersion));
+		report.setReportDate(entityEncode(reportDate));
+		report.setReportVersion(entityEncode(reportVersion));
+		report.setDesc(entityEncode(reportDesc));
+
+		String description = Constant.messages.getString("exportReport.details.description");
```

Comment:
i18n string references should be in lower case (not camelCaps), and separated by periods. So `exportreport.details.description` would be the preferred form (since the component/extension is exportreport as a name not separate elements).

Assume the same throughout..